expression/bench_test.go

```diff
@@ -218,8 +219,26 @@ type defaultRandGen struct {
 	*rand.Rand
 }
 
+type lockedSource struct {
```

Comment:
why add this struct? seems not used in other changed files.

Reply:
get it, as some tests are run in parallal, so need make newDefaultRandGen safe for concurrent